src/java/org/apache/cassandra/config/ValidatedPropertyLoader.java

```diff
@@ -0,0 +1,117 @@
+    {
+        this.nonListenableProperty = nonListenableProperty;
+    }
+
+    @Override
+    protected Property createPropertyForField(Class<?> root, Field field)
+    {
+        Property property = new FieldProperty(field);
+        if (nonListenableProperty.test(property))
+            return property;
+
+        ListenableProperty<?, ?> listenable = new ListenableProperty<>(property);
+        if (property.getAnnotation(ValidatedBy.class) != null)
+            listenable.addBeforeHandler(ValidationPropertyHandler.create(root, listenable.delegate()));
+        return listenable;
+    }
+
+    private static class ValidationPropertyHandler<S, T> implements ListenableProperty.Handler<S, T>
+    {
+        private final Method method;
+        private final String name;
+
+        private ValidationPropertyHandler(Method method, String name)
+        {
+            this.method = method;
+            this.name = name;
+        }
+
+        public static <S, T> ValidationPropertyHandler<S, T> create(Class<?> root, Property property)
+        {
+            Preconditions.checkNotNull(property.getAnnotation(ValidatedBy.class));
+            ValidatedBy annotation = property.getAnnotation(ValidatedBy.class);
+            Class<?> clazz = FBUtilities.classForName(annotation.useClass(), "validate method");
+            Class<?>[] searchPattern = ArrayUtils.addAll(new Class<?>[]{ root, property.getType() });
+            for (Method refMethod : clazz.getDeclaredMethods())
+            {
+                int modifiers = refMethod.getModifiers();
+                if (Modifier.isPrivate(modifiers) || Modifier.isProtected(modifiers))
+                    continue;
+
+                if (refMethod.getName().equals(annotation.useClassMethod()) &&
+                    Arrays.equals(refMethod.getParameterTypes(), searchPattern) &&
+                    (refMethod.getReturnType() == Void.TYPE || refMethod.getReturnType().equals(property.getType())))
+                {
+                    return new ValidationPropertyHandler<>(refMethod, property.getName());
+                }
+            }
+            throw new ConfigurationException(String.format("Required method '%s(%s)' not found for field '%s'",
+                                                           annotation.useClassMethod(),
+                                                           Arrays.stream(searchPattern)
+                                                                 .map(Class::getCanonicalName)
+                                                                 .collect(Collectors.joining(", ")),
+                                                           property.getName()), false);
+        }
+
+        /** {@inheritDoc} */
+        @SuppressWarnings("unchecked")
+        @Override
+        public T handle(S source, T oldValue, T newValue)
+        {
+            try
+            {
+                Object result = method.invoke(source, source, newValue);
```

Comment:
this isn't correct, should have `null` as the first argument, as you don't have access to `DatabaseDescriptor`

Reply:
Also, you only searched for `Parent, Value`, so this should fail as the arguments you pass in are different

Reply:
Patterns added here:
https://github.com/apache/cassandra/pull/2334#discussion_r1205939691

The `null` is used, as you mentioned about it here.